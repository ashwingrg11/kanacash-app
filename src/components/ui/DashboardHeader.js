import React from 'react';
import {View, StyleSheet} from 'react-native';
import theme from '~/components/theme/Style';
import {SemiBoldText, BoldText} from '~/components/ui/Text';
import {SendMoney} from '~/components/ui/Icon';

export default function DashboardHeader({name}) {
  return (
    <View style={styles.wrapper}>
      <View style={styles.containerStyle}>
        <SemiBoldText text={`Hi, ${name}`} invert />
      </View>
      <View style={styles.bntWrapper}>
        <BoldText text="Send Money" invert style={styles.sendMoneyText} />
        <SendMoney />
      </View>
    </View>
  );
}

DashboardHeader.defaultProps = {
  name: 'John Doe',
};

const styles = StyleSheet.create({
  wrapper: {
    alignItems: 'center',
  },
  containerStyle: {
    height: 180,
    width: '100%',
    backgroundColor: theme.black,
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  bntWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.red,
    padding: 20,
    borderRadius: 5,
    marginTop: -30,
    flexDirection: 'row',
    // position: 'absolute',
    // top: '80%',
  },
  sendMoneyText: {
    marginRight: 8,
  },
});
