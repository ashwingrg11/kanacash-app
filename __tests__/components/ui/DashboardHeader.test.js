import 'react-native';
import React from 'react';

import DashboardHeader from '~/components/ui/DashboardHeader';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<DashboardHeader/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DashboardHeader name="Joe Doe"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
