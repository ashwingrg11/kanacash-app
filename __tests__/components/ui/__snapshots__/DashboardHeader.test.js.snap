// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardHeader/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1F1F1F",
        "borderRadius": 5,
        "height": 180,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      invert={true}
      style={
        Array [
          Object {
            "fontFamily": "OpenSans-SemiBold",
            "fontSize": 20,
            "lineHeight": 25,
          },
          Object {
            "color": "#fff",
          },
          undefined,
        ]
      }
      text="Hi, Joe Doe"
    >
      Hi, Joe Doe
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#B31F31",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": -30,
        "padding": 20,
      }
    }
  >
    <Text
      invert={true}
      style={
        Array [
          Object {
            "fontFamily": "OpenSans-Bold",
            "fontSize": 24,
            "fontWeight": "600",
            "lineHeight": 41,
          },
          Object {
            "color": "#fff",
          },
          Object {
            "marginRight": 8,
          },
        ]
      }
      text="Send Money"
    >
      Send Money
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="30"
      bbWidth="30"
      fill="none"
      focusable={false}
      height="30"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 30,
            "width": 30,
          },
        ]
      }
      vbHeight={30}
      vbWidth={30}
      width="30"
      xmlns="http://www.w3.org/2000/Svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M30 26.25H3.75V24.375H28.125V11.25H30V26.25Z"
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M24.375 7.5V20.625H1.875V7.5H24.375ZM26.25 5.625H0V22.5H26.25V5.625Z"
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M5.625 11.25H3.75V16.875H5.625V18.75H13.125C11.8818 18.75 10.6895 18.2561 9.81044 17.3771C8.93136 16.498 8.4375 15.3057 8.4375 14.0625C8.4375 12.8193 8.93136 11.627 9.81044 10.7479C10.6895 9.86886 11.8818 9.375 13.125 9.375H5.625V11.25Z"
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M20.625 11.25V9.375H13.125C14.3682 9.375 15.5605 9.86886 16.4396 10.7479C17.3186 11.627 17.8125 12.8193 17.8125 14.0625C17.8125 15.3057 17.3186 16.498 16.4396 17.3771C15.5605 18.2561 14.3682 18.75 13.125 18.75H20.625V16.875H22.5V11.25H20.625Z"
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
